compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -414,6 +415,20 @@ class ReplDriver(settings: Array[String],
       }
       state
 
+    case Settings(arg) => arg match
+      case "" =>
+        given ctx: Context = state.context
+        for (s <- ctx.settings.userSetSettings(ctx.settingsState).sortBy(_.name))
+          out.println(s"${s.name} = ${if s.value == "" then "\"\"" else s.value}")
+        state
+      case _  =>
+        setup(tokenize(arg).toArray, rootCtx) match
+          case Some((files, ictx)) =>
```

Comment:
`files` is unused, so naturally I wonder what if it is non-empty. Scala 2 ignores it. The other issue with settings is that they can be cached. I noticed that Scala 2 `:se -Vtyper` turns on the setting but to no effect. Ideally it would reset as much as necessary for the request to actually do something useful. Not sure about supplying files, but maybe just say "ignoring spurious arguments $files".